common/ucf-degree-common.php

```diff
@@ -321,6 +321,135 @@ private static function update_profile( $post_id, $result ) {
 			return false;
 		}
 
+		/**
+		 * Add a tuition exception for a degree
+		 * @author Jim Barnes
+		 * @since 3.2.0
+		 * @param int $post_id The post id
+		 */
+		public static function add_tuition_exception( $post_id, $value ) {
+			$plan_code    = get_post_meta( $post_id, 'degree_plan_code', true );
+			$subplan_code = get_post_meta( $post_id, 'degree_subplan_code', true );
+
+			# Setup the params
+			$params = array(
+				'plan_code'    => $plan_code,
+				'subplan_code' => $subplan_code
+			);
+
+			$base_url       = UCF_Degree_Config::get_option_or_default( 'api_base_url' );
+			$key            = UCF_Degree_Config::get_option_or_default( 'api_key' );
+			$update_tuition = UCF_Degree_Config::get_option_or_default( 'update_tuition' );
+
+			// Return out if update_tuition is false or url or key are not set.
+			if ( ! $base_url || ! $key || ! $update_tuition ) return;
+
+			$url = $base_url . 'tuition-mappings/';
+
+			# Get first result from search.
+			$results = self::fetch_api_values( $url, $params );
+
+			if ( ! $results ) {
+				$status = 'retrieval-error';
+
+				add_filter( 'redirect_post_location', function( $location ) use ( $status ) {
+					return add_query_arg( 'degree_tuition_status', $status, $location );
+				} );
+			}
+
+			# Make sure the result is correct
+			$existing = null;
+			if ( count( $results ) > 0 ) {
+				$existing = self::return_verified_result( $results, $params );
+			}
+
+			if ( $existing ) {
+				self::update_existing_exception( $post_id, $existing, $base_url, $key, $value );
+			} else {
+				// Only create a new exception if the skip value is true
+				if ( $value ) {
+					self::create_new_exception( $post_id, $plan_code, $subplan_code, $key, $base_url );
+				}
+			}
+		}
+
+		/**
+		 * Helper function for updating existing tuition exceptions.
+		 * @author Jim Barnes
+		 * @since 3.2.0
+		 * @param int $post_id The post id to update
+		 * @param object $result The API record to update
+		 * @param string $key The API key
+		 * @param string $base_url The base URL of the API
+		 */
```

Comment:
Description for `$value` is missing; `$existing` is incorrectly labeled as `$result` here